botnim/query.py

```diff
@@ -187,3 +233,43 @@ def format_mapping(mapping: Dict, indent: int = 0) -> str:
     
     return "\n".join(result)
 
+def elastic_vector_search_handler(environment: str, bot_name: str, context_name: str, query: str, num_results: int = None) -> str:
+    """
+    Handles Elasticsearch vector search requests from the assistant
+    
+    Args:
+        environment (str): Environment to use (production or staging)
+        bot_name (str): Name of the bot to use
+        context_name (str): Name of the context to search in
+        query (str): The search query text
+        num_results (int, optional): Number of results to return, or None to use context default
+        
+    Returns:
+        str: Formatted search results as a string
+    """
+    # If num_results is None, get the default from context config
```

Comment:
Copying comment from other #34:

`QueryClient` already loads the config and has the context - it should handle managing the default `num_results` extraction in case it's None.
There's absolutely no need for loading here the config.yaml as well.

Reply:
ok, removing the elastic search handler then.